.github/workflows/ci.yml

```diff
@@ -0,0 +1,104 @@
+name: infoware CI
+on:
+  push:
+  pull_request:
+
+jobs:
+  linux-x64:
+    name: Linux x64 (${{matrix.cfg.cpp-version}})
+    runs-on: ${{matrix.cfg.os}}
+    strategy:
+      fail-fast: false
+      matrix:
+        cfg:
+          - { os: ubuntu-18.04, cpp-version: g++-8 }
+          - { os: ubuntu-18.04, cpp-version: g++-9 }
+          - { os: ubuntu-18.04, cpp-version: clang++-7 }
+          - { os: ubuntu-18.04, cpp-version: clang++-8 }
+          - { os: ubuntu-18.04, cpp-version: clang++-9 }
+
+    steps:
+      - name: Checkout infoware
+        uses: actions/checkout@v2
+        with:
+          path: main
```

Comment:
Why?

Reply:
I don't know which part your asking for so I'll explain like this:
A lot of projects might just say `- uses: actions/checkout@v2` which is a perfectly fine way to go about it, though I prefer to add the `name` tag, as that adds a label to what is being checked-out in the Github Actions UI, verses just seeing the action name.

I prefer to add `path: main` encase in the future, lets say more things are required to test, and other repos need to get checked out, it lets you add another `actions/checkout` step without having to modify the existing steps.

I understand both of my explanations start with "I prefer", and the project name may not want to set things up in the same manner. Let me know what you would prefer, I'll be happy to make the change.

Reply:
Having the clone path not be the repository basename definitely took me aback a bit, yeah, I'd much rather this be `infoware` (which I assume would be the default?)

Reply:
The default is no additional directory. It will be directly in the `${{github.workspace}}`. I'd switch it to infoware. If you'd prefer no directory, and the root in the workspace, I can do that too.

Reply:
`$PWD/infoware` is good, guess I'm unfamiliar with this particular set of defaults

Reply:
Got it!